src/App.tsx

```diff
@@ -9,39 +9,44 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './app/store';
+import { fetchUsers } from './features/users/usersSlice';
+import { setAuthor } from './features/author/authorSlice';
+import { setSelectedPost } from './features/selectedPost/selectedPostSlice';
+import { fetchPosts, clearPosts } from './features/posts/postsSlice';
```

Comment:
Check that `fetchPosts` and `clearPosts` are correctly defined in the `postsSlice` file.